openedx/core/storage.py

```diff
@@ -90,6 +91,24 @@ def __init__(self, acl=None, bucket=None, custom_domain=None, **settings):
         super(S3ReportStorage, self).__init__(acl=acl, bucket=bucket, **settings)
 
 
+@deconstructible
+class OverwriteStorage(FileSystemStorage):
+    """
+    Comes from http://www.djangosnippets.org/snippets/976/
+    (even if it already exists in S3Storage for ages)
+    See also Django #4339, which might add this functionality to core.
```

Comment:
When I started something similar a while ago, I tried to add some context to the docstring:

> FileSystemStorage subclass which automatically overwrites any previous file with the same name; used in test runs to avoid test file proliferation.  Copied from django-storages when this class was removed in version 1.6.

Otherwise my branch was essentially identical to what you have here.